docs/docs/building-with-components.md

```diff
@@ -2,9 +2,192 @@
 title: Building with Components
 ---
 
-Rough outline
+## Requirements
 
-* Docs expect basic familiarity with React
-* All Gatsby site UI is built with React components.
-* Concise description of why components amazing model for building sites.
-* Perhaps link to other good articles on "Why components are amazing".
+You'll need to have basic understanding of React components.
+
+The [official tutorial](https://facebook.github.io/react/tutorial/tutorial.html) is a good place to start.
+
+## Why React components?
+
+React allows to create big complicated webapps while preserving modularity, reusabilty, and clear abstraction. React also has great devtools available for debugging the applicattion.
+
+Here you can read more about [Thinking in React](https://facebook.github.io/react/docs/thinking-in-react.html).
+
+## How does Gatsby use React Components?
+
+Everything in Gatsby is built using Components.
+
+A basic directory structure of a project looks like this:
+
+```sh
+.
+├── gatsby-config.js
+├── package.json
+└── src
+    ├── html.jsx
+    ├── pages
+    │   ├── index.jsx
+    │   └── posts
+    │       ├── 01-01-2017
+    │       │   └── index.md
+    │       ├── 01-02-2017
+    │       │   └── index.md
+    │       └── 01-03-2017
+    │           └── index.md
+    ├── templates
+    │   └── post.jsx
+    │
+    └── layouts
+        └── index.jsx
+```
+
+### Base HTML generation
+
+`src/html.jsx` is responsible for generating the base HTML document, where the react component tree is going to be mounted.
+
+Example:
+
+```jsx
+import React from 'react';
+import favicon from './favicon.png';
+
+let inlinedStyles = '';
+if (process.env.NODE_ENV === 'production') {
+  try {
+    inlinedStyles = require('!raw-loader!../public/styles.css');
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+export default class HTML extends React.Component {
+  render() {
+    let css;
+    if (process.env.NODE_ENV === 'production') {
+      css = (
+        <style
+          id="gatsby-inlined-css"
+          dangerouslySetInnerHTML={{ __html: inlinedStyles }}
+        />
+      );
+    }
+    return (
+
+      <html lang="en">
+        <head>
+          <meta charSet="utf-8" />
+          <meta
+            name="viewport"
+            content="width=device-width, initial-scale=1.0"
+          />
+          {this.props.headComponents}
+          <link rel="shortcut icon" href={favicon} />
+          {css}
+        </head>
+        <body>
+          <div
+            id="___gatsby"
+            dangerouslySetInnerHTML={{ __html: this.props.body }}
+          />
+          {this.props.postBodyComponents}
+        </body>
+      </html>
+    );
+  }
+}
+```
+
+In this file you can modify the `<head>` metadata, general structure of the document and add external links.
+
+### UI Layouts
+
+`src/layouts/index.jsx` (optional) is a wrapper component for everything generated by Gatsby. You can use it to create consistent layouts between pages.
```

Comment:
"wraps page components. You can use use it for site-wide parts of the site like headers and footers."